bootstrap/005_table_migrations.sql

```diff
@@ -25,35 +25,62 @@ BEGIN
-      RAISE OUT_OF_SYNC;
-  END IF;
 
-  let columns string := (
-      SELECT LISTAGG('"' || COLUMN_NAME || '" ' || DATA_TYPE, ', ') WITHIN GROUP (ORDER BY ORDINAL_POSITION)
+  let columns_to_drop string := (
+      SELECT LISTAGG('"' || COLUMN_NAME || '" ' , ', ')
       FROM (
-      SELECT COLUMN_NAME, DATA_TYPE, ORDINAL_POSITION FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = :view_schema AND TABLE_NAME = :view_name
+      SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = :table_schema AND TABLE_NAME = :table_name
       MINUS
-      SELECT COLUMN_NAME, DATA_TYPE, ORDINAL_POSITION FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = :table_schema AND TABLE_NAME = :table_name
+      SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = :view_schema AND TABLE_NAME = :view_name
       )
 
       );
-  let alter_statement string := 'ALTER TABLE "' || :table_schema || '"."' || :table_name || '" ADD ' || columns;
-  if (columns <> '') then
-      execute immediate alter_statement;
+
+  let alter_statement string := 'ALTER TABLE "' || :table_schema || '"."' || :table_name || '"';
+
+  let alter_table_add_column string := '';
+  let alter_table_drop_column string := '';
+
+  if (columns_to_add <> '') then
+    alter_table_add_column := ' ADD ' || columns_to_add;
+    execute immediate alter_statement || alter_table_add_column;
+  end if;
+
+  if (columns_to_drop <> '') then
+    alter_table_drop_column := ' DROP ' || columns_to_drop;
+    execute immediate alter_statement || alter_table_drop_column;
+  end if;
+
+  if (columns_to_add <> '' OR columns_to_drop <> '') then
       SYSTEM$LOG_INFO('Migration executed for ' || :view_schema || '.' || :view_name || ' and ' || :table_schema || '.' || :table_name);
-      SYSTEM$ADD_EVENT('table altered', {'alter_statement': alter_statement });
-      RETURN columns;
+      SYSTEM$ADD_EVENT('table altered', {'alter_statement': alter_statement || alter_table_add_column || alter_table_drop_column });
+      RETURN alter_table_add_column || alter_table_drop_column;
   else
     SYSTEM$LOG_INFO('No migration need for ' || :view_schema || '.' || :view_name || ' and ' || :table_schema || '.' || :table_name);
     RETURN null;
   end if;
 END;
+
+
+CREATE OR REPLACE PROCEDURE internal.migrate_view()
+    RETURNS STRING
+    LANGUAGE SQL
+    COMMENT = 'Re-create view used in the App. This is required when Snowflake adds a new column to query_history, or removes an existing column from query_history'
+    AS
+BEGIN
+    call INTERNAL.create_view_QUERY_HISTORY_COMPLETE_AND_DAILY();
+    call INTERNAL.create_view_enriched_query_history();
+    call INTERNAL.create_view_enriched_query_history_daily();
+    call INTERNAL.create_view_enriched_query_history_hourly();
+    call INTERNAL.create_view_LABELED_QUERY_HISTORY();
```

Comment:
Does this take into account existing label columns? Or do we have to re run labellign also?

Reply:
Good point. I should call `UPDATE_LABEL_VIEW()` which will also re-create `Labeled_query_history` view.